### ISSUE TITLE
Client-side load balancer

### ISSUE THREAD

**Issue by trustin** [line/armeria#87]
For the time being, simple round-robin load balancer will be enough to test if our client API design needs a change for client-side load balancing support.

Consider how we could provide great user experience when configuring a load-balanced client. For example, a user could use builder pattern to provide a list of remote hosts, weights and other parameters.

One implementation tactic I thought about is to define a custom name resolver:

```
new StaticServerGroup().add("hostA", portA).add(...).add(...)
        .strategy(ROUND_ROBIN).register("myGroup");
Clients.newClient("tbinary+http://myGroup/...", ...);
```

/cc @anuraaga @inch772

**Comment by shelocks** [comment]
Is anybody working on this?Otherwise i would like to take this one.

**Comment by trustin** [comment]
@shelocks It's all yours. Let me look forward to your PR :-)

**Comment by imasahiro** [comment]
Think we can close this issue for now since the PR was already merged and released at 0.18.0.Final
